ecs-agent/wsclient/client.go

```diff
@@ -190,18 +206,21 @@ type MakeRequestHookFunc func([]byte) ([]byte, error)
 // Connect opens a connection to the backend and upgrades it to a websocket. Calls to
 // 'MakeRequest' can be made after calling this, but responses will not be
 // receivable until 'Serve' is also called.
-func (cs *ClientServerImpl) Connect() error {
+func (cs *ClientServerImpl) Connect(disconnectMetricName string,
```

Comment:
No, it would not always be metrics.ACSDisconnectTimeoutMetricName. Since we are moving this periodic timeout logic to websocket library, which is used by both ACS and TACS, we need to pass the disconnect metric name to specifiy which connection was disconnected, i.e. ACS or TACS. There is no agent-specific aspect here, but more to differentiate between diff connections which use the wsclient library.